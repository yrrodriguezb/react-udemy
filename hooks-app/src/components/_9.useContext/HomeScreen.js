import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const HomeScreen = () => {

  const { user } = useContext(UserContext);


  return (
    <div className="p-5">
      <h1>HomeScreen</h1>
      <hr />

      <pre className='container'>
        { JSON.stringify(user, null, 4) }
      </pre>
    </div>
   
  )
}
