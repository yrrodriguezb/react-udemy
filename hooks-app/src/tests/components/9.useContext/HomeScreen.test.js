import React from 'react'
import { mount } from 'enzyme'
import '@testing-library/jest-dom'
import { HomeScreen } from '../../../components/_9.useContext/HomeScreen'
import { UserContext } from '../../../components/_9.useContext/UserContext'


describe('Pruebas en <HomeScreen />', () => {
  const user = {
    name: 'Yeison Roriguez',
    email: 'yrrodriguez@gmail.com'
  }

  const wrapper = mount(
    <UserContext.Provider value={ { user } }>
      <HomeScreen />
    </UserContext.Provider>
  );
  
  test('Debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  })
})