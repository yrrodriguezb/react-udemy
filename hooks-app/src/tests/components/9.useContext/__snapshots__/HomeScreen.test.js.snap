// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen /> Debe mostrarse correctamente 1`] = `
<div
  className="p-5"
>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre
    className="container"
  >
    {
    "name": "Yeison Roriguez",
    "email": "yrrodriguez@gmail.com"
}
  </pre>
</div>
`;
